OneSignalExample/Assets/OneSignal/src/OneSignal.cs

```diff
@@ -146,761 +207,1007 @@ public class OSEmailSubscriptionStateChanges {
-         result.action = action;
-
-         builder.notificationOpenedDelegate(result);
-      }
-   }
-      
-   // Called from the native SDK - Called when device is registered with onesignal.com service or right after IdsAvailable
-   //                          if already registered.
-   private void onIdsAvailable(string jsonString) {
-      if (idsAvailableDelegate != null) {
-         var ids = Json.Deserialize(jsonString) as Dictionary<string, object>;
-         idsAvailableDelegate((string)ids["userId"], (string)ids["pushToken"]);
-      }
-   }
+            string response = jsonObject["response"] as string;
 
-   // Called from the native SDK - Called After calling GetTags(...)
-   private void onTagsReceived(string jsonString) {
-      if (tagsReceivedDelegate != null)
-         tagsReceivedDelegate(Json.Deserialize(jsonString) as Dictionary<string, object>);
-   }
+            if (delegates.ContainsKey(delegateIdSuccess)) {
+                var logoutEmailSuccessDelegate = (OnLogoutEmailSuccess) delegates[delegateIdSuccess];
+                delegates.Remove(delegateIdSuccess);
+                delegates.Remove(delegateIdFailure);
+                logoutEmailSuccessDelegate();
+            }
+        }
 
-   // Called from the native SDK
-   private void onPostNotificationSuccess(string response) {
-      if (postNotificationSuccessDelegate != null) {
-         OnPostNotificationSuccess tempPostNotificationSuccessDelegate = postNotificationSuccessDelegate;
-         postNotificationFailureDelegate = null;
-         postNotificationSuccessDelegate = null;
-         tempPostNotificationSuccessDelegate(Json.Deserialize(response) as Dictionary<string, object>);
-      }
-   }
+        // Called from the native SDK
+        private void onLogoutEmailFailure(string jsonString) {
+            if (string.IsNullOrEmpty(jsonString))
+                return;
 
-   // Called from the native SDK
-   private void onSetEmailSuccess() {
-      if (setEmailSuccessDelegate != null) {
-         OnSetEmailSuccess tempSuccessDelegate = setEmailSuccessDelegate;
-         setEmailSuccessDelegate = null;
-         setEmailFailureDelegate = null;
-         tempSuccessDelegate();
-      }
-   }
+            // Break part the jsonString which might contain a 'delegate_id' and a 'response'
+            var jsonObject = Json.Deserialize(jsonString) as Dictionary<string, object>;
 
-   // Called from the native SDK
-   private void onSetEmailFailure(string error) {
-      if (setEmailFailureDelegate != null) {
-         OnSetEmailFailure tempFailureDelegate = setEmailFailureDelegate;
-         setEmailFailureDelegate = null;
-         setEmailSuccessDelegate = null;
-         tempFailureDelegate(Json.Deserialize(error) as Dictionary<string, object>);
-      }
-   }
+            // Check if the delegate should be processed
+            if (!prcoessSuccessFailureDelegate(jsonObject))
+                return;
 
-   // Called from the native S
```

Comment:
This has duplicated logic from `processDelegate`, could this be called from here?